rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+
+Please note that `improvement:` is **not** supported, there is an [open pull request](https://github.com/commitizen/conventional-commit-types/pull/16) to remove it from [`commitizen`](https://github.com/commitizen/cz-cli).
+
+If we use [`git revert`](https://git-scm.com/docs/git-revert) we will use the [`--no-commit`](https://git-scm.com/docs/git-revert#Documentation/git-revert.txt---no-commit) flag if we're reverting a `BREAKING CHANGE` because this would also be a breaking change, or, if we're reverting a new feature. We will then use `commitizen` to write a commit that includes a reference to the `BREAKING CHANGE`.
+
+eg.
+> revert(decimal): remove on change handler 
+>
+> The on change hander was implemented incorrectly and caused major regressions.  
+> We're reverting this commit in favour of PR #1.
+>
+> This commit reverts #abcdef  
+> - feat(decimal): add on change handler
+>
+>BREAKING CHANGE: decimal onChange prop has been removed
+
+## Learning Git
+
+It's important that developers have a good working knowledge of Git. Here are a collection of resources that I have
+personally used:
+
+- [Atlassian Tutorials](https://www.atlassian.com/git/tutorials/what-is-version-control)
+- [Pro Git Book](https://git-scm.com/book/en/v2)
+- [Learn Git Branching](https://learngitbranching.js.org/)
+- [GitHub Labs](https://lab.github.com/)
+
+### Atomic Commits
+
+> an atomic commit is an operation that applies a set of distinct changes as a single operation  
+> [[Source]](https://en.wikipedia.org/wiki/Atomic_commit) 
+
+Atomic commits help us create frequent small releases. It makes code review and cherry-picking/reverting commits easier.
+
+Writing an atomic commits may mean re-writing your **local** history before making a PR. I recommend reading both
+[Atlassian rewriting history](https://www.atlassian.com/git/tutorials/rewriting-history) and [Git book rewriting history](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
+
+> #### Don’t push your work until you’re happy with it
+> One of the cardinal rules of Git is that, since so much work is local within your clone, you have a great deal of freedom to rewrite your history **locally**. However, once you push your work, it is a different story entirely, and you should consider pushed work as final unless you have good reason to change it. In short, you should avoid pushing your work until you’re happy with it and ready to share it with the rest of the world.  
+> [[Source]](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)
+
+> So, before you run git rebase, always ask yourself, “Is anyone else looking at this branch?” If the answer is yes, take your hands off the keyboard and start thinking about a non-destructive way to make your changes (e.g., the git revert command). Otherwise, you’re safe to re-write history as much as you like.  
```

Comment:
```suggestion
> So, before you run `git rebase`, always ask yourself, “Is anyone else looking at this branch?” If the answer is yes, take your hands off the keyboard and start thinking about a non-destructive way to make your changes (e.g., the `git revert` command). Otherwise, you’re safe to re-write history as much as you like.  
```